packages/ui/src/forms/Form/fieldReducer.ts

```diff
@@ -284,20 +284,38 @@ export function fieldReducer(state: FormState, action: FieldAction): FormState {
         // ..
         // This is a performance enhancement for saving
         // large documents with hundreds of fields
-        const newState = {}
+        const newState: FormState = {}
 
-        Object.entries(action.state).forEach(([path, field]) => {
+        for (const [path, newField] of Object.entries(action.state)) {
           const oldField = state[path]
-          const newField = field
+
+          if (newField.valid !== false) {
+            newField.valid = true
+          }
+          if (newField.passesCondition !== false) {
+            newField.passesCondition = true
+          }
 
           if (!dequal(oldField, newField)) {
             newState[path] = newField
           } else if (oldField) {
             newState[path] = oldField
           }
-        })
+        }
+
         return newState
       }
+
+      if (action.sanitize) {
+        for (const field of Object.values(action.state)) {
```

Comment:
Can you add a `@todo` block here for us to remove this in the next major release?